### PR TITLE
[NEXUS-8549,NEXUS-8518,NEXUS-8609] purge legacy security, add default anonymous roles, remove PluginIdentity

### PR DESCRIPTION
**Pull request by jdillon**
https://issues.sonatype.org/browse/NEXUS-8549
https://issues.sonatype.org/browse/NEXUS-8518
https://issues.sonatype.org/browse/NEXUS-8609

related:

https://github.com/sonatype/nexus-pro/pull/543

includes:

https://github.com/sonatype/nexus-oss/pull/1325

... with modifications to make JS permissions work exactly like Java WildcardPermission

### CODE REVIEW

**Comment by jdillon** [comment]
http://bamboo.s/browse/NX3-OSSF292-2 :white_check_mark:

**Comment by jdillon** [comment]
Seems I accidentally mixed in some of the wildcard perm in the UI changes, so should merge that PR down first, then I can resolve here.

File: plugins/rapture/nexus-coreui-plugin/src/main/java/org/sonatype/nexus/coreui/UserRoleMappingsXO.groovy [hunk 1]
```diff
@@ -23,11 +29,14 @@ import org.hibernate.validator.constraints.NotEmpty
...
+  @NotBlank
+  @UserExists(groups = Create)
   String userId
 
-  @NotEmpty
+  @NotBlank
+  @RealmExists(groups = Create)
   String realm
 
+  @RolesExist(groups = [Create, Update])
```
Comment (adreghiciu) [inline plugins/rapture/nexus-coreui-plugin/src/main/java/org/sonatype/nexus/coreui/UserRoleMappingsXO.groovy]: Is not necessary to set create/update groups as in this case the Default group will be used

**Comment by adreghiciu** [comment]
+1

**Comment by jdillon** [comment]
manually merged

**Pull request closed without merging**